mlflow/store/artifact/gcs_artifact_repo.py

```diff
@@ -131,5 +125,23 @@ def _download_file(self, remote_file_path, local_path):
             remote_full_path, chunk_size=self._GCS_DOWNLOAD_CHUNK_SIZE
         ).download_to_filename(local_path, timeout=self._GCS_DEFAULT_TIMEOUT)
 
+    def _parse_mlflow_gcs_vars(self, environment_variable, var_type):
+        if var_type == "timeout":
+            if environment_variable:
+                if environment_variable == -1:
+                    return None
+                else:
+                    return float(environment_variable)
+            else:
+                from google.cloud.storage.constants import _DEFAULT_TIMEOUT
+
+                return _DEFAULT_TIMEOUT
+
+        elif var_type == "chunk_size":
+            if environment_variable:
+                return int(environment_variable)
+            else:
+                return None
```

Comment:
Nit: I think it was clearer / easier to reason about how the `self._GCS_UPLOAD/DOWNLOAD_CHUNK_SIZE` and `self._GCS_DEFAULT_TIMEOUT` values were being set before this function was introduced. We seem to be pushing special case logic for each value into a single function, which IMO makes things a bit harder to interpret / maintain.

Can we remove this function and instead perform the `-1` -> `None` conversion for `self._GCS_DEFAULT_TIMEOUT` as:

```
self._GCS_DEFAULT_TIMEOUT = (
    float(os.environ.get("MLFLOW_GCS_DEFAULT_TIMEOUT"))
    if os.environ.get("MLFLOW_GCS_DEFAULT_TIMEOUT")
    else _DEFAULT_TIMEOUT
)
# If the user-supplied timeout environment variable value is -1, use `None` for `self._GCS_DEFAULT_TIMEOUT`
# to disable timeouts
self._GCS_DEFAULT_TIMEOUT = None if self._GCS_DEFAULT_TIMEOUT == -1
```